inc/plugins/class-dynamic-content.php

```diff
@@ -649,6 +662,84 @@ public function get_link( $data ) {
 		return apply_filters( 'otter_blocks_evaluate_dynamic_content_link', '', $data );
 	}
 
+	/**
+	 * Mark the exceptions for Dynamic request.
+	 *
+	 * Features that are using complex functions like `get_content` (directly or indirectly) might not behave correctly in every situation.
+	 * Based on the context, we decide if it is safe to use those functions..
+	 *
+	 * @param array $data Dynamic request.
+	 * @return array Dynamic request with marked exceptions.
+	 */
+	public function mark_exceptions( $data ) {
+		if ( 'postExcerpt' === $data['type'] || 'postContent' === $data['type'] ) {
+			if (
+				'valid' === apply_filters( 'product_neve_license_status', false ) &&
+				class_exists( '\Neve_Pro\Modules\Custom_Layouts\Module' )
+			) {
+				$post = get_post( $data['context'] );
+				if ( isset( $post ) && 'neve_custom_layouts' === $post->post_type ) {
+					$key = $this->get_exception_key( $data, $post->ID );
+					if ( $key ) {
+						$data[ $key ] = true;
+					}
+				}
+			}
+
+			if ( 'postContent' === $data['type'] ) {
+				$content = get_the_content( $data['context'] );
+				if ( strpos( $content, 'data-type="postContent"' ) ) {
+					$key = $this->get_exception_key( $data, $post->ID );
+					if ( $key ) {
+						$data[ $key ] = true;
+					}
+				}
+			}
```

Comment:
This should prevent the following situation:
<img width="1513" alt="Screenshot 2023-04-26 at 15 18 44" src="https://user-images.githubusercontent.com/17597852/234575068-2bf9dcb8-8f1a-4cb2-9230-7348ed21a36b.png">

We will not process that still has the dynamic tag since it triggers the infinite loop.